src/backend/distributed/deparser/ruleutils_13.c

```diff
@@ -1003,7 +1003,8 @@ set_relation_column_names(deparse_namespace *dpns, RangeTblEntry *rte,
 	 * get_rte_attribute_name, except that it's important to disregard dropped
 	 * columns.  We put NULL into the array for a dropped column.
 	 */
-	if (rte->rtekind == RTE_RELATION)
+	if (rte->rtekind == RTE_RELATION ||
+        GetRangeTblKind(rte) == CITUS_RTE_SHARD)
```

Comment:
As in other deparser functions, we could do;
```suggestion
	int rtekind = GetRangeTblKind(rte);
	if (rtekind == CITUS_RTE_RELATION ||
        rtekind == CITUS_RTE_SHARD)
```

But effectively both are checking for the same thing.

Reply:
I searched for `GetRangeTblKind` instances in ruleutils_15.c, and in all, except one instance, we directly do comparisons from `GetRangeTblKind` function. So, I vote for the current choice, not the comment's suggestion.

Reply:
Besides doing the comparison with GetRangeTblKind(), I'm also wondering whether we should compare `rte->rtekind with RTE_RELATION`, or `GetRangeTblKind(rte) with CITUS_RTE_RELATION`.

Both should mean the same, latter one sounds slightly more intuitive.

Reply:
Hmm now I see your point. That one exception instance is actually the only instance that is similar to this one: where we put both `RTE_RELATION` and `CITUS_RTE_SHARD` under the same umbrella. In that case, I am changing my vote :D up to you to decide.